transport/awssqs/publisher.go

```diff
@@ -0,0 +1,128 @@
+package awssqs
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/go-kit/kit/endpoint"
+)
+
+// Publisher wraps an sqs client and queue, and provides a method that
+// implements endpoint.Endpoint.
+type Publisher struct {
+	sqsClient        SQSClient
+	queueURL         *string
+	responseQueueURL *string
+	enc              EncodeRequestFunc
+	dec              DecodeResponseFunc
+	before           []PublisherRequestFunc
+	after            []PublisherResponseFunc
+	timeout          time.Duration
+}
+
+// NewPublisher constructs a usable Publisher for a single remote method.
+func NewPublisher(
+	sqsClient SQSClient,
+	queueURL *string,
```

Comment:
Why is this a pointer to a string?

Reply:
This is because the aws sdk uses string pointers. I'll change them to strings and call a pointer to the var when needed